src/library/scala/collection/ArrayOps.scala

```diff
@@ -1637,4 +1637,12 @@ final class ArrayOps[A](private val xs: Array[A]) extends AnyVal {
     *  @return `true` if this array has `that` as a suffix, `false` otherwise.
     */
   def endsWith[B >: A](that: Iterable[B]): Boolean = mutable.ArraySeq.make(xs).endsWith(that)
+
+  /** Ensure we are providing an array of the correct element type.
+   */
+  private def narrowly(ys: Array[_]): Array[A] =
```

Comment:
is this cast safe? It seems like this `Array[_ <: A]` because we know we can assign A from an argument?

Can we add a comment explaining the argument of why this cast is safe? I wonder if we don't need `implicitly[ClassTag[A]].runtimeClass == ys.getClass.getComponentType` for this to be safe.